docs/8egress.adoc

```diff
@@ -1,174 +1,165 @@
 ----
 
-Note: it will not work now but it will once Istio-ized
+_Important:_ We have a 3rd Deployment to manage the v3 version of recommendation. 
 
 [source,bash]
 ----
-ctrl-c
+oc apply -f <(istioctl kube-inject -f ../../kubernetes/Deployment-v3.yml) -n tutorial
 
-docker ps | grep egress
+oc get pods -w
+----
 
-oc apply -f <(istioctl kube-inject -f src/main/kubernetes/Deployment.yml) -n istioegress
+==== OpenShift S2I strategy (if you DON'T have access to Docker daemon)
 
-oc create -f src/main/kubernetes/Service.yml
+[source, bash]
+----
+mvn clean package -f recommendation/java/vertx
+oc new-app -l app=recommendation,version=v3 --name=recommendation-v3 --context-dir=recommendation/java/vertx -e JAEGER_SERVICE_NAME=recommendation JAEGER_ENDPOINT=http://jaeger-collector.istio-system.svc:14268/api/traces JAEGER_PROPAGATION=b3 JAEGER_SAMPLER_TYPE=const JAEGER_SAMPLER_PARAM=1 JAVA_OPTIONS='-Xms128m -Xmx256m -Djava.net.preferIPv4Stack=true' fabric8/s2i-java~https://github.com/redhat-developer-demos/istio-tutorial -o yaml  > recommendation-v3.yml
+oc apply -f <(istioctl kube-inject -f recommendation-v3.yml) -n tutorial
+oc cancel-build bc/recommendation-v3
+oc delete svc/recommendation-v3
+oc start-build recommendation-v3 --from-dir=. --follow
+----
 
-oc expose service egressgithub
+==== Wait for v3 to be deployed
 
-curl egressgithub-istioegress.$(minishift ip).nip.io
+Wait for those pods to show "2/2", the istio-proxy/envoy sidecar is part of that pod
 
-cd ../..
+[source,bash]
 ----
+NAME                                  READY     STATUS    RESTARTS   AGE
+customer-3600192384-fpljb             2/2       Running   0          17m
+preference-243057078-8c5hz           2/2       Running   0          15m
+recommendation-v1-60483540-9snd9     2/2       Running   0          12m
+recommendation-v2-2815683430-vpx4p   2/2       Running   0          15s
+recommendation-v3-7b445dd469-j6rkg   2/2       Running   0          2m
+----
+
+cd ../../..
 
 === Istio-ize Egress
 
-[source,bash]
-----
-istioctl create -f istiofiles/egress_httpbin.yml -n istioegress
+Be sure you do not have any previous destination rule nor virtual service installed.
 
-istioctl get serviceentry
+Let's redirect all traffic to reccomendation:v3.
 
-curl egresshttpbin-istioegress.$(minishift ip).nip.io
+[source, bash]
 ----
-
-or shell into the pod by getting its name and then using that name with oc exec
-
-[source,bash]
+istioctl create -f istiofiles/destination-rule-recommendation-v1-v2-v3.yml -n tutorial
+istioctl create -f istiofiles/virtual-service-recommendation-v3.yml
 ----
-oc exec -it $(oc get pods -o jsonpath="{.items[*].metadata.name}" -l app=egresshttpbin,version=v1) -c egresshttpbin /bin/bash
-
-curl localhost:8080
 
-curl httpbin.org/user-agent
+Then access to the service:
 
-curl httpbin.org/headers
+[source, bash]
+----
+curl customer-tutorial.$(minishift ip).nip.io
```

Comment:
Add a note that it might take more than 10 seconds to reply with "upstream request timeout"

Reply:
Maybe you can add a curl -m 5 or -m 10 to specify a timeout in seconds. Having the endpoint without an answer for so long made me wonder if it was something wrong.